core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1919,9 +1966,8 @@ private[zio] trait ZIOFunctions extends Serializable {
       r <- ZIO.runtime[R]
       a <- ZIO.uninterruptibleMask { restore =>
             val f = register(k => r.unsafeRunAsync_(k.to(p)))
-            restore(f.catchAllCause(p.halt)).fork.flatMap { f =>
-              restore(p.await).onInterrupt(f.interrupt)
-            }
+
+            restore(f.catchAllCause(p.halt)).fork *> restore(p.await) //.onInterrupt(f.interrupt)
```

Comment:
Comment unneeded, right? `f` should get interrupted automatically

Reply:
Fixed.